radio/src/gui/colorlcd/menu_radio.cpp

```diff
@@ -35,17 +35,43 @@
 RadioMenu::RadioMenu():
   TabsGroup(ICON_RADIO)
 {
+  build();
+}
+
+RadioMenu::~RadioMenu()
+{
+  storageCheck(true);
+}
+
+void RadioMenu::build()
+{
+  _radioThemesEnabled = radioThemesEnabled();
+  _radioGFEnabled = radioGFEnabled();
+  _radioTrainerEnabled = radioTrainerEnabled();
+
   addTab(new RadioToolsPage());
   addTab(new RadioSdManagerPage());
   addTab(new RadioSetupPage());
-  addTab(new ThemeSetupPage());
-  addTab(new SpecialFunctionsPage(g_eeGeneral.customFn));
-  addTab(new RadioTrainerPage());
+  if (_radioThemesEnabled)
+    addTab(new ThemeSetupPage());
+  if (_radioGFEnabled)
+    addTab(new SpecialFunctionsPage(g_eeGeneral.customFn));
+  if (_radioTrainerEnabled)
+    addTab(new RadioTrainerPage());
   addTab(new RadioHardwarePage());
   addTab(new RadioVersionPage());
 }
 
-RadioMenu::~RadioMenu()
+void RadioMenu::checkEvents()
 {
-    storageCheck(true);
+  TabsGroup::checkEvents();
+
+  if (_radioThemesEnabled != radioThemesEnabled() ||
+      _radioGFEnabled != radioGFEnabled() ||
+      _radioTrainerEnabled != radioTrainerEnabled()) {
+    removeAllTabs();
+    build();
+    setCurrentTab(0);
+    setCurrentTab(2);
```

Comment:
Does this need both `setCurrentTab` calls or is 74 a copy paste artifact?

Reply:
Yes, both are needed. I've added a comment for future reference.